empress/support_files/js/empress.js

```diff
@@ -897,34 +900,84 @@ define([
         return keyInfo;
     };
 
-    /**
-     * Finds the branches that are unique to each category in obs
+    /*
+     * Projects the groups in obs up the tree.
      *
-     * @param {Object} observations grouped by categories
+     * This function performs two distinct operations:
+     *      1) Removes the non-unique observations from each group in obs
+     *         (i.e. performs an 'exclusive or' between each group) and, if
+     *         addNonUnique is true, then places all non-unique observations
+     *         in a "non-unique" group.
      *
-     * @return {Object} the branches of the tree that are unique to category in
-                        obs
-    */
-    Empress.prototype._projectObservations = function (obs) {
-        var tree = this._tree;
-        var categories = Object.keys(obs);
+     *      2) Assigns internal nodes to a group if all of its children belong
+     *         to the same group.
+     *
+     * Note: All tips that are not passed into obs are considered to belong to
+     *       a "not-represented" group.
+     *       Also, if addNonUnique is true, then a "non-unique" object will
+     *       be inserted into the return object.
+     *
+     * @param {Object} obs Maps cateogires to a set of observations (i.e. tips)
+     * @param {Boolean} addNonUnique If true, and then insert the "non-unique"
+     *                               group into the returning object.
+     * @return {Object}
+     */
+    Empress.prototype._projectObservations = function (obs, addNonUnique) {
+        var tree = this._tree,
+            categories = Object.keys(obs),
+            notRepresented = new Set(),
+            i,
+            j;
+
+        // find "non-represented" tips
+        // Note: the following uses postorder traversal
+        for (i = 1; i < tree.size; i++) {
+            if (tree.isleaf(tree.postorderselect(i))) {
+                var represented = false;
+                for (j = 0; j < categories.length; j++) {
+                    if (obs[categories[j]].has(i)) represented = true;
+                }
+                if (!represented) notRepresented.add(i);
+            }
+        }
 
         // assign internal nodes to appropriate category based on children
         // iterate using postorder
```

Comment:
```suggestion
        // iterate using postorder
        // Note that, although we don't explicitly iterate over the
        // root (at index tree.size) in this loop, we iterate over all its
        // descendants; so in the event that all leaves are unique,
        // the root can still get assigned to a group.
```

(I had to double-check this to make sure it worked properly in this case, so this should save future work for other people with the same question :)